custom/icds_reports/mpr_sqldata.py

```diff
@@ -0,0 +1,2307 @@
+                    'alias': 'M_sev_mod_resident_weighted'
+                },
+                {
+                    'columns': ('F_sev_mod_resident_weighted', 'M_sev_mod_resident_weighted'),
+                    'alias': 'all_sev_resident_weighted'
+                }
+            ),
+            (
+                '',
+                '%',
+                {
+                    'columns': ('F_sev_resident_weighed_count',),
+                    'func': div,
+                    'second_value': 'F_resident_weighed_count',
+                },
+                {
+                    'columns': ('M_sev_resident_weighed_count',),
+                    'func': div,
+                    'second_value': 'M_resident_weighed_count',
+                },
+                {
+                    'columns': ('F_sev_resident_weighed_count_1',),
+                    'func': div,
+                    'second_value': 'F_resident_weighed_count_1',
+                },
+                {
+                    'columns': ('M_sev_resident_weighed_count_1',),
+                    'func': div,
+                    'second_value': 'M_resident_weighed_count_1',
+                },
+                {
+                    'columns': ('F_sev_resident_weighed_count_2',),
+                    'func': div,
+                    'second_value': 'F_resident_weighed_count_2',
+                },
+                {
+                    'columns': ('M_sev_resident_weighed_count_2',),
+                    'func': div,
+                    'second_value': 'M_resident_weighed_count_2',
+                },
+                {
+                    'columns': ('F_sum_sev_resident_weighted',),
+                    'func': div,
+                    'second_value': 'resident_female_weight'
+                },
+                {
+                    'columns': ('M_sum_sev_resident_weighted',),
+                    'func': div,
+                    'second_value': 'resident_male_weight'
+                },
+                {
+                    'columns': ('all_sev_resident_weighted',),
+                    'func': div,
+                    'second_value': 'all_resident_weight'
+                },
+            )
+        )
+
+
+class MPRImmunizationCoverage(ICDSMixin, MPRData):
+
+    title = '9. Immunization Coverage'
+    slug = 'immunization_coverage'
+
+    @property
+    def subtitle(self):
+        if self.config['location_id']:
+            data = self.custom_data(selected_location=self.selected_location, domain=self.config['domain'])
+            children_completing = data.get('open_child_count', 0)
+            vaccination = data.get('open_child_1yr_immun_complete', 1)
+            immunization = "%.2f%%" % ((children_completing / (vaccination or 1)) * 100)
+            return (
+                '(I) Children Completing 12 months during the month: {0}'.format(children_completing),
+                '(II) Of this, number of children who have rec
```

Comment:
http://stackoverflow.com/questions/10678229/how-can-i-selectively-escape-percent-in-python-strings